lib/mixpanel/context.tsx

```diff
@@ -1,113 +1,88 @@
+  children: React.ReactNode;
+  trackingService: TrackingService;
+  defaultEventContext?: WebMixpanelEvent['context'] | MobileMixpanelEvent['context'];
+  disableSessionStorage?: boolean;
 }
 
 const MixpanelContext = createContext<MixpanelContextProps | null>(null);
 
 export function useMixpanelContext() {
-    const context = useContext(MixpanelContext);
+  const context = useContext(MixpanelContext);
 
-    if (!context) {
-        throw new Error('<MixpanelProvider /> not found');
-    }
+  if (!context) {
+    throw new Error('<MixpanelProvider /> not found');
+  }
 
-    return context;
+  return context;
 }
 
 export function MixpanelProvider({
-    children,
-    eventApiClient,
-    defaultEventContext,
+  children,
+  trackingService,
+  defaultEventContext,
+  disableSessionStorage = false,
 }: MixpanelProviderProps) {
-    const [eventContext, setEventContext] = useState<MixpanelEvent['context']>(
-        defaultEventContext || {}
-    );
-
-    const trackEvent = useCallback(
-        (event: MixpanelEvent) => {
-            // only send events on the client
-            if (typeof window === 'undefined') {
-                return;
-            }
-
-            const utmParams = extractUtmParams(window.location.search);
-
-            eventApiClient({
-                ...event,
-                context: {
-                    title: document.title,
-                    pathname: window.location.pathname,
-                    pwa: isStandalonePWA(),
-                    ...eventContext,
-                    ...utmParams,
-                    ...event.context,
-                },
-            }).catch((e) => console.error(e));
+  const [eventContext, setEventContext] = useState<WebMixpanelEvent['context'] | MobileMixpanelEvent['context']>(
+    defaultEventContext || {}
+  );
+
+  const trackEvent = useCallback(
+    (event: WebMixpanelEvent | MobileMixpanelEvent) => {
+      trackingService.trackEvent({
+        ...event,
+        context: {
+          ...eventContext,
+          ...event.context,
         },
-        [eventApiClient, eventContext]
-    );
-
-    const trackPageView = useCallback(
-        (event: MixpanelPageViewEvent) => {
-            // only send events on the client
-            if (typeof window === 'undefined') {
-                return;
-            }
-
-            const utmParams = extractUtmParams(window.location.search);
-
-            eventApiClient({
-                ...event,
-                name: 'Page view',
-                context: {
-                    pwa: isStandalonePWA(),
-                    ...utmParams,
-                    ...event.context,
-                },
-            }).catch((e) => console.error(e));
+      });
+    },
+    [trackingService, eventContext]
+  );
+
+  const trackPageView = useCallback(
+    (event: WebMixpanelPageViewEvent | MobileMixpanelPageViewEvent) => {
+      trackingService.trackPageView({
+        ...event,
+        context: {
+          ...eventContext,
```

Comment:
Ik had deze eerder bewust niet gezet bij PageViewEvents. dus je kon alleen een defaultEventContext zetten voor de  trackEvent functie. 

Maar mogelijk is er wel een use case dat je dit ook voor de pageviews wilt gebruiken. Moeten alleen wel uitkijken dat er geen nieuwe page view getriggert wordt op het moment dat je die defaultEventContext gaat aanpassen...